packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -0,0 +1,422 @@
+    const chunk = chunks[i];
+    try {
+      const chunkResults = (await aggregateTransactionsAndCall(multicall.address, web3, chunk)) as {
+        ancillaryData: string;
+      }[];
+      results.push(...chunkResults);
+    } catch (error) {
+      for (let j = 0; j < chunk.length; j++) {
+        const call = chunk[j];
+        try {
+          const market = markets[i * chunkSize + j];
+          const adapter = market.resolvedBy === params.binaryAdapterAddress ? binaryAdapter : ctfAdapter;
+          const result = await adapter.callStatic.questions(market.questionID);
+          results.push(result);
+        } catch {
+          results.push({ ancillaryData: "0x" });
+          console.log("Failed to get ancillary data for market", call);
+        }
+      }
+    }
+  }
+
+  return markets.map((market, index) => {
+    return {
+      ...market,
+      ancillaryData: results[index].ancillaryData,
+    };
+  });
+};
+
+export const getPolymarketOrderBooks = async (
+  params: MonitoringParams,
+  markets: PolymarketWithEventData[],
+  networker: NetworkerInterface
+): Promise<(PolymarketWithEventData & MarketOrderbooks)[]> => {
+  const results: (PolymarketWithEventData & MarketOrderbooks)[] = [];
+
+  await processMarketsInChunks(markets, 30, async (marketChunk: PolymarketWithEventData[]) => {
+    const chunkResults = await Promise.all(
+      marketChunk.map(async (market) => {
+        const marketOne = market.clobTokenIds[0];
+        const marketTwo = market.clobTokenIds[1];
+        const apiUrlOne = params.apiEndpoint + `/book?token_id=${marketOne}`;
+        const apiUrlTwo = params.apiEndpoint + `/book?token_id=${marketTwo}`;
+        const { bids: outcome1Bids, asks: outcome1Asks } = (await networker.getJson(apiUrlOne, {
+          method: "get",
+        })) as PolymarketOrderBook;
+
+        const { bids: outcome2Bids, asks: outcome2Asks } = (await networker.getJson(apiUrlTwo, {
+          method: "get",
+        })) as PolymarketOrderBook;
+
+        const stringToNumber = (orderBook: {
+          bids: {
+            price: string;
+            size: string;
+          }[];
+          asks: {
+            price: string;
+            size: string;
+          }[];
+        }) => {
+          return {
+            bids: orderBook.bids.map((bid) => {
+              return {
+                price: Number(bid.price),
+                size: Number(bid.size),
+              };
+            }),
+            asks: orderBook.asks.map((ask) => {
+              return {
+                price: Number(ask.price),
+                size: Number(ask.size),
+              };
+            }),
+          };
+        };
+
+        return {
+          ...market,
+          ...({
+            orderBooks: [
+              stringToNumber({ bids: outcome1Bids || [], asks: outcome1Asks || [] }),
+              stringToNumber({ bids: outcome2Bids || [], asks: outcome2Asks || [] }),
+            ],
+          } as MarketOrderbooks),
```

Comment:
nit: why is this cast needed?

I ask because the types are somewhat hard to reason about here, so I want to make sure the compiler is verifying.